libbeat/monitoring/report/buffer/reporter.go

```diff
@@ -0,0 +1,119 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package buffer
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/monitoring"
+)
+
+// reporter is a struct that will fill a ring buffer for each monitored registry.
+type reporter struct {
+	config
+	wg         sync.WaitGroup
+	done       chan struct{}
+	registries map[string]*monitoring.Registry
+
+	// ring buffers for namespaces
+	entries map[string]*ringBuffer
+}
+
+// MakeReporter creates and starts a reporter with the given config.
+func MakeReporter(beat beat.Info, cfg *common.Config) (*reporter, error) {
+	config := defaultConfig()
+	if cfg != nil {
+		if err := cfg.Unpack(&config); err != nil {
+			return nil, err
+		}
+	}
+
+	r := &reporter{
+		config:     config,
+		done:       make(chan struct{}),
+		registries: map[string]*monitoring.Registry{},
+		entries:    map[string]*ringBuffer{},
+	}
+
+	for _, ns := range r.config.Namespaces {
+		reg := monitoring.GetNamespace(ns).GetRegistry()
+		r.registries[ns] = reg
+		r.entries[ns] = newBuffer(r.config.Size)
+	}
+
+	r.wg.Add(1)
+	go func() {
+		defer r.wg.Done()
+		r.snapshotLoop()
+	}()
+	return r, nil
+}
+
+// Stop will stop the reporter from collecting new information.
+// It will not clear any previously collected data.
+func (r *reporter) Stop() {
+	close(r.done)
+	r.wg.Wait()
+	// Clear entries?
```

Comment:
Remove comment so there is no confusion on the godoc of the function, no need to clear the ring buffer there.